internalv2compat/compat.go

```diff
@@ -63,3 +64,19 @@ type IsHTTP interface {
 type IsThrift interface {
 	isThrift()
 }
+
+var (
+	v2TracingEnabledMutex sync.RWMutex
+	v2TracingEnabled      bool
```

Comment:
use [`atomic.Bool`](https://pkg.go.dev/sync/atomic#Bool) instead?

Reply:
I considered it but picked a mutex because this should only be referenced during setup/init, and I feel like the benefit of a memory barrier outweighs the performance benefits of atomics.